src/vsr/grid.zig

```diff
@@ -1177,10 +1177,22 @@ pub fn GridType(comptime Storage: type) type {
         pub fn next_batch_of_block_requests(grid: *Grid, requests: []vsr.BlockRequest) usize {
             assert(grid.callback != .cancel);
             assert(requests.len > 0);
+            assert(requests.len == constants.grid_repair_reads_max);
 
             // Prioritize requests for blocks with stalled Grid reads, so that commit/compaction can
-            // continue.
-            const request_faults_count = @min(grid.read_global_queue.count, requests.len);
+            // continue. We divide the buffer up between `read_global_queue` and
+            // `blocks_missing.faulty_blocks` so that we always request blocks from both queues.
+            // Surplus from `blocks_missing.faulty_blocks` may be used by `read_global_queue`.
+            const request_faults_count_max =
+                @divFloor(requests.len, 2) +
+                (@divFloor(requests.len, 2) -| grid.blocks_missing.faulty_blocks.count());
```

Comment:
Ah, if `requests.len` is an odd number, then `request_faults_count_max` will at most be `requests.len - 1`!
We could switch one of the `@divFloor`'s to a `div_ceil`, but the intent might be clearer with e.g.:

```
requests.len - @min(
  @divFloor(requests.len, 2),
  grid.blocks_missing.faulty_blocks.count(),
);
```

Reply:
Ah, that is a great point regarding odd-numbered requests.len, especially because there's no asserts stopping us from setting `grid_repair_reads_max` to an odd number!